cirq-core/cirq/transformers/gauge_compiling/gauge_compiling.py

```diff
@@ -128,10 +126,45 @@ def sample(self, gate: ops.Gate, prng: np.random.Generator) -> ConstantGauge:
             post_q0=self.post_q0,
             post_q1=self.post_q1,
             swap_qubits=self.swap_qubits,
-            support_sweep=self.support_sweep,
         )
 
 
+@frozen
+class TwoQubitGateSymbolizer:
+    """Parameterizes two qubit gates with symbols.
+
+    Attributes:
+        symbolizer: A callable that takes a two-qubit gate and a sequence of symbols,
+            and returns a tuple containing the parameterized gate and a dictionary
+            mapping symbol names to their values.
+        n_symbols: The number of symbols to use for parameterization.
+    """
+
+    symbolizer_fn: Callable[
+        [ops.Gate, Sequence[sympy.Symbol]], Tuple[ops.Gate, Dict[str, Union[float, int]]]
```

Comment:
Nit - here and at other instances of `Union[float, int]`

```suggestion
        [ops.Gate, Sequence[sympy.Symbol]], Tuple[ops.Gate, Dict[str, numbers.Real]]
```

Reply:
Good point! Done.

In switching to Real, extra code type hinting using `isinstance(..., Real)` is necessary as `Types from "numbers" aren't supported for static type checking`. So I wonder what is the typical way to determine whether to use `float` (should be good enough in most cases as int can naturally be converted to float?), `Real` (better generic real number support, easy to switch to Complex?), `Union[float, int]` (type checking friendly) in cirq?

Reply:
@babacry - which `isinstace(..., Real)` do you have in mind?  The b2fbfb532f1d4a7756748d12239547dc5eb68920 seems to add

https://github.com/quantumlib/Cirq/blob/b2fbfb532f1d4a7756748d12239547dc5eb68920/cirq-core/cirq/transformers/gauge_compiling/sqrt_cz_gauge.py#L71-L72

but that one should be necessary, because CZPowGate.exponent can be also sympy.Expr.

As for the second question, I would opt for float for a variable that should not receive numpy scalars.  Otherwise, numbers.Real will cover int, float, and all related numpy numeric types.